convert2rhel/utils.py

```diff
@@ -119,24 +119,22 @@ def restart_system():
     from convert2rhel.toolopts import tool_opts
 
     if tool_opts.restart:
-        run_subprocess("reboot")
+        run_subprocess(["reboot"])
     else:
         loggerinst.warning("In order to boot the RHEL kernel, restart of the system is needed.")
 
 
-def run_subprocess(cmd="", print_cmd=True, print_output=True):
+def run_subprocess(cmd, print_cmd=True, print_output=True):
     """Call the passed command and optionally log the called command (print_cmd=True) and its
     output (print_output=True). Switching off printing the command can be useful in case it contains
     a password in plain text.
+
+    The cmd is specified as a list starting with the command and followed by a list of arguments.
+    Example: ["dnf", "repoquery", "kernel"]
     """
     if print_cmd:
```

Comment:
If we do, then we probably want to check whether this is a string rather than checking if it is not a list to preserve duck-typing (subprocess.Popen should work with anything that is a Sequence [a tuple, for instance.])

A question, though:
* Are we guarding against str being passed in because str used to be passed in?
  * If so, we probably want to limit the time that we perform this check.
  * If not, what is the difference between this function and other functions which take lists and would error if given a string?

Reply:
I think you're right, we should compare against `str` and not `list`, which makes more sense. 

Regarding the question, I thought about checking against this because of what you mentioned in your gist. Someone could forget to use `shlex.quote()` when passing an `str` to our `run_subprocess()` and thus, we would be in the same spot as of right now. 

Let me know what you think about that, it's just a suggestion (don't know either if I understood correctly what I read 😅)